File: 63 Name/python/name.py
"""
NAME

simple string manipulations on the user's name

Ported by Dave LeCompte
"""


def print_with_tab(space_count, msg):
    if space_count > 0:
        spaces = " " * space_count
    else:
        spaces = ""
    print(spaces + msg)


def is_yes_ish(answer):
    cleaned = answer.strip().upper()
    if cleaned == "Y" or cleaned == "YES":
        return True
    return False


def main():
    print_with_tab(34, "NAME")
    print_with_tab(15, "CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY")
    print()
    print()
    print()
    print("HELLO.")
    print("MY NAME iS CREATIVE COMPUTER.")
    name = input("WHAT'S YOUR NAME (FIRST AND LAST)?")
    print()
    name_as_list = list(name)
    reversed_name = "".join(name_as_list[::-1])
    print(f"THANK YOU, {reversed_name}.")
    print()
    print("OOPS!  I GUESS I GOT IT BACKWARDS.  A SMART")
    print("COMPUTER LIKE ME SHOULDN'T MAKE A MISTAKE LIKE THAT!")
    print()
    print()
    print("BUT I JUST NOTICED YOUR LETTERS ARE OUT OF ORDER.")

    sorted_name = "".join(sorted(name_as_list))
    print(f"LET'S PUT THEM IN ORDER LIKE THIS: {sorted_name}")
    print()
    print()

    print("DON'T YOU LIKE THAT BETTER?")
    like_answer = input()
    print()
    if is_yes_ish(like_answer):
        print("I KNEW YOU'D AGREE!!")
    else:
        print("I'M SORRY YOU DON'T LIKE IT THAT WAY.")
    print()
    print(f"I REALLY ENJOYED MEETING YOU, {name}.")
    print("HAVE A NICE DAY!")


main()

File: 63 Name/javascript/name.js
// NAME
//
// Converted from BASIC to Javascript by Oscar Toledo G. (nanochess)
//

function print(str)
{
    document.getElementById("output").appendChild(document.createTextNode(str));
}

function input()
{
    var input_element;
    var input_str;
    
    return new Promise(function (resolve) {
                       input_element = document.createElement("INPUT");
                       
                       print("? ");
                       input_element.setAttribute("type", "text");
                       input_element.setAttribute("length", "50");
                       document.getElementById("output").appendChild(input_element);
                       input_element.focus();
                       input_str = undefined;
                       input_element.addEventListener("keydown", function (event) {
                                                      if (event.keyCode == 13) {
                                                      input_str = input_element.value;
                                                      document.getElementById("output").removeChild(input_element);
                                                      print(input_str);
                                                      print("\n");
                                                      resolve(input_str);
                                                      }
                                                      });
                       });
}

function tab(space)
{
    var str = "";
    while (space-- > 0)
        str += " ";
    return str;
}

var str;
var b;

// Main program
async function main()
{
    print(tab(34) + "NAME\n");
    print(tab(15) + "CREATIVE COMPUTING  MORRISTOWN, NEW JERSEY\n");
    print("\n");
    print("\n");
    print("\n");
    print("HELLO.\n");
    print("MY NAME IS CREATIVE COMPUTER.\n");
    print("WHAT'S YOUR NAME (FIRST AND LAST)");
    str = await input();
    l = str.length;
    print("\n");
    print("THANK YOU, ");
    for (i = l; i >= 1; i--)
        print(str[i - 1]);
    print(".\n");
    print("OOPS!  I GUESS I GOT IT BACKWARDS.  A SMART\n");
    print("COMPUTER LIKE ME SHOULDN'T MAKE A MISTAKE LIKE THAT!\n");
    print("\n");
    print("BUT I JUST NOTICED YOUR LETTERS ARE OUT OF ORDER.\n");
    print("LET'S PUT THEM IN ORDER LIKE THIS: ");
    b = [];
    for (i = 1; i <= l; i++)
        b[i - 1] = str.charCodeAt(i - 1);
    b.sort();
    for (i = 1; i <= l; i++)
        print(String.fromCharCode(b[i - 1]));
    print("\n");
    print("\n");
    print("DON'T YOU LIKE THAT BETTER");
    ds = await input();
    if (ds == "YES") {
        print("\n");
        print("I KNEW YOU'D AGREE!!\n");
    } else {
        print("\n");
        print("I'M SORRY YOU DON'T LIKE IT THAT WAY.\n");
    }
    print("\n");
    print("I REALLY ENJOYED MEETING YOU " + str + ".\n");
    print("HAVE A NICE DAY!\n");
}

main();
